src/ParallelAlgorithms/Interpolation/Actions/CleanUpInterpolator.hpp

```diff
@@ -87,10 +90,12 @@ struct CleanUpInterpolator {
           if constexpr (std::is_same_v<
                             typename InterpolationTargetTag::temporal_id,
                             typename Tag::temporal_id>) {
-            const auto& found =
+            const auto& finished_temporal_ids =
                 get<Vars::HolderTag<Tag, Metavariables>>(holders)
                     .temporal_ids_when_data_has_been_interpolated;
-            if (found.count(temporal_id) == 0) {
+            if (std::find(finished_temporal_ids.begin(),
+                          finished_temporal_ids.end(),
+                          temporal_id) == finished_temporal_ids.end()) {
```

Comment:
Include `Utilities/Algorithm.hpp` and try

```cpp
alg::found(finished_temporal_ids, temporal_id) {
```

I think it looks cleaner. Same for the other places in this commit.